### ISSUE TITLE
NameError: name 'EmptyPyArrowIterator' is not defined - v 2.1.0 on Windows

### ISSUE THREAD

**Issue by sakkammadam** [snowflakedb/snowflake-connector-python#249]
Please answer these questions before submitting your issue. Thanks!

1. What version of Python are you using (`python --version`)? **3.6.8**

2. What operating system and processor architecture are you using (`python -c 'import platform; print(platform.platform())'`)? **Windows-10-10.0.17763-SP0**

3. What are the component versions in the environment (`pip list`)?  
Package                    Version
-------------------------- ----------
```
asn1crypto                 1.2.0
azure-common               1.1.23
azure-storage-blob         2.1.0
azure-storage-common       2.1.0
boto3                      1.10.33
botocore                   1.13.33
cdm-base-utils             1.0
cdm-db-utils               1.0
certifi                    2019.11.28
cffi                       1.13.2
chardet                    3.0.4
cryptography               2.8
docutils                   0.15.2
future                     0.18.2
idna                       2.8
ijson                      2.5.1
jmespath                   0.9.4
numpy                      1.17.4
oscrypto                   1.1.0
pandas                     0.25.3
pip                        18.1
pycparser                  2.19
pycryptodomex              3.9.4
PyJWT                      1.7.1
pyOpenSSL                  19.1.0
PyPika                     0.35.16
python-dateutil            2.8.0
pytz                       2019.3
requests                   2.22.0
s3transfer                 0.2.1
setuptools                 40.6.2
six                        1.13.0
snowflake-connector-python 2.1.0
sqlparse                   0.3.0
urllib3                    1.25.7
```

4. What did you do?
If possible, provide a recipe for reproducing the error.
A complete runnable program is good.
I created a cursor and I tried to run a query
 **blah = crs.execute("SELECT A.ID FROM STAGE.INQUIRIES_CLN A JOIN STAGE.IAU_INQUIRIES_CLN B ON A.ID = B.ID")**

5. What did you expect to see?
Results of query in blah

6. What did you see instead?

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\sakkammadam\.pyenv\pyenv-win\versions\3.6.8-amd64\lib\site-packages\snowflake\connector\cursor.py", line 580, in execute
    self._init_result_and_meta(data, _use_ijson)
  File "C:\Users\sakkammadam\.pyenv\pyenv-win\versions\3.6.8-amd64\lib\site-packages\snowflake\connector\cursor.py", line 630, in _init_result_and_meta
    self._result = ArrowResult(data, self, use_dict_result=self._use_dict_result)
  File "arrow_result.pyx", line 42, in snowflake.connector.arrow_result.ArrowResult.__init__
  File "arrow_result.pyx", line 156, in snowflake.connector.arrow_result.ArrowResult._reset
NameError: name 'EmptyPyArrowIterator' is not defined

```

7. Can you set logging to DEBUG and collect the logs?

```
import logging
import os

for logger_name in ['snowflake.sqlalchemy', 'snowflake.connector', 'botocore']: 
    logger = logging.getLogger(logger_name)
    logger.setLevel(logging.DEBUG)
    ch = logging.StreamHandler()
    ch.setLevel(logging.DEBUG)
    ch.setFormatter(logging.Formatter('%(asctime)s - %(threadName)s %(filename)s:%(lineno)d - %(funcName)s() - %(levelname)s - %(message)s'))
    logger.addHandler(ch)
```
Here you go - it is making the connection and the query results  
2019-12-12 17:22:31,253 - MainThread cursor.py:464 - execute() - DEBUG - executing SQL/command
2019-12-12 17:22:31,253 - MainThread cursor.py:487 - execute() - DEBUG - binding: [SELECT A.ID FROM STAGE.INQUIRIES_CLN A JOIN STAGE.IAU_INQUIRIES_CLN B ON A.ID = ...] with input=[None], processed=[{}]
2019-12-12 17:22:31,253 - MainThread cursor.py:519 - execute() - INFO - query: [SELECT A.ID FROM STAGE.INQUIRIES_CLN A JOIN STAGE.IAU_INQUIRIES_CLN B ON A.ID = ...]
2019-12-12 17:22:31,254 - MainThread connection.py:1068 - _next_sequence_counter() - DEBUG - sequence counter: 5
2019-12-12 17:22:31,254 - MainThread cursor.py:344 - _execute_helper() - DEBUG - running query [SELECT A.ID FROM STAGE.INQUIRIES_CLN A JOIN STAGE.IAU_INQUIRIES_CLN B ON A.ID = ...]
2019-12-12 17:22:31,254 - MainThread cursor.py:353 - _execute_helper() - DEBUG - is_file_transfer: False
2019-12-12 17:22:31,254 - MainThread connection.py:824 - cmd_query() - DEBUG - _cmd_query
2019-12-12 17:22:31,255 - MainThread connection.py:846 - cmd_query() - DEBUG - sql=[SELECT A.ID FROM STAGE.INQUIRIES_CLN A JOIN STAGE.IAU_INQUIRIES_CLN B ON A.ID = ...], sequence_id=[5], is_file_transfer=[None]
2019-12-12 17:22:31,255 - MainThread network.py:940 - _use_requests_session() - DEBUG - Active requests sessions: 1, idle: 0
2019-12-12 17:22:31,255 - MainThread network.py:639 - _request_exec_wrapper() - DEBUG - remaining request timeout: None, retry cnt: 1
2019-12-12 17:22:31,256 - MainThread network.py:778 - _request_exec() - DEBUG - socket timeout: 60
2019-12-12 17:22:31,385 - MainThread ssl_wrap_socket.py:413 - ssl_wrap_socket_with_ocsp() - DEBUG - OCSP Mode: FAIL_OPEN, OCSP response cache file name: None
2019-12-12 17:22:31,386 - MainThread ocsp_snowflake.py:422 - reset_ocsp_response_cache_uri() - DEBUG - ocsp_response_cache_uri: file://C:/Users/sakkammadam/AppData/Local/Snowflake/Caches/ocsp_response_cache.json
2019-12-12 17:22:31,386 - MainThread ocsp_snowflake.py:424 - reset_ocsp_response_cache_uri() - DEBUG - OCSP_VALIDATION_CACHE size: 78
2019-12-12 17:22:31,387 - MainThread ocsp_snowflake.py:218 - reset_ocsp_dynamic_cache_server_url() - DEBUG - OCSP response cache server is enabled: http://ocsp.snowflakecomputing.com/ocsp_response_cache.json
2019-12-12 17:22:31,387 - MainThread ocsp_snowflake.py:252 - reset_ocsp_dynamic_cache_server_url() - DEBUG - OCSP dynamic cache server RETRY URL: None
2019-12-12 17:22:31,387 - MainThread ocsp_snowflake.py:563 - check_ocsp_response_cache_lock_dir() - DEBUG - The cache is older than 1 day. Deleted the cache file: C:/Users/sakkammadam/AppData/Local/Snowflake/Caches/ocsp_response_cache.json
2019-12-12 17:22:31,387 - MainThread ocsp_snowflake.py:465 - read_ocsp_response_cache_file() - DEBUG - Failed to locate OCSP response cache file. No worry. It will validate with OCSP server: C:/Users/sakkammadam/AppData/Local/Snowflake/Caches/ocsp_response_cache.js
on
2019-12-12 17:22:31,388 - MainThread ocsp_snowflake.py:972 - validate() - DEBUG - validating certificate: cdm.snowflakecomputing.com
2019-12-12 17:22:31,388 - MainThread ocsp_asn1crypto.py:354 - extract_certificate_chain() - DEBUG - # of certificates: 4
2019-12-12 17:22:31,389 - MainThread ocsp_asn1crypto.py:361 - extract_certificate_chain() - DEBUG - subject: OrderedDict([('common_name', '*.snowflakecomputing.com')]), issuer: OrderedDict([('country_name', 'US'), ('organization_name', 'Amazon'), ('organizational_u
nit_name', 'Server CA 1B'), ('common_name', 'Amazon')])
2019-12-12 17:22:31,390 - MainThread ocsp_asn1crypto.py:361 - extract_certificate_chain() - DEBUG - subject: OrderedDict([('country_name', 'US'), ('organization_name', 'Amazon'), ('organizational_unit_name', 'Server CA 1B'), ('common_name', 'Amazon')]), issuer: Ord
eredDict([('country_name', 'US'), ('organization_name', 'Amazon'), ('common_name', 'Amazon Root CA 1')])
2019-12-12 17:22:31,391 - MainThread ocsp_asn1crypto.py:361 - extract_certificate_chain() - DEBUG - subject: OrderedDict([('country_name', 'US'), ('organization_name', 'Amazon'), ('common_name', 'Amazon Root CA 1')]), issuer: OrderedDict([('country_name', 'US'), ('
state_or_province_name', 'Arizona'), ('locality_name', 'Scottsdale'), ('organization_name', 'Starfield Technologies, Inc.'), ('common_name', 'Starfield Services Root Certificate Authority - G2')])
2019-12-12 17:22:31,393 - MainThread ocsp_asn1crypto.py:361 - extract_certificate_chain() - DEBUG - subject: OrderedDict([('country_name', 'US'), ('state_or_province_name', 'Arizona'), ('locality_name', 'Scottsdale'), ('organization_name', 'Starfield Technologies,
Inc.'), ('common_name', 'Starfield Services Root Certificate Authority - G2')]), issuer: OrderedDict([('country_name', 'US'), ('organization_name', 'Starfield Technologies, Inc.'), ('organizational_unit_name', 'Starfield Class 2 Certification Authority')])
2019-12-12 17:22:31,395 - MainThread ocsp_asn1crypto.py:383 - create_pair_issuer_subject() - DEBUG - not found issuer_der: OrderedDict([('country_name', 'US'), ('organization_name', 'Starfield Technologies, Inc.'), ('organizational_unit_name', 'Starfield Class 2 Ce
rtification Authority')])
2019-12-12 17:22:31,396 - MainThread ocsp_snowflake.py:588 - find_cache() - DEBUG - hit cache for subject: OrderedDict([('common_name', '*.snowflakecomputing.com')])
2019-12-12 17:22:31,399 - MainThread ocsp_asn1crypto.py:201 - is_valid_time() - DEBUG - Verifying the attached certificate is signed by the issuer. Valid Not After: 2022-10-24 20:13:22+00:00
2019-12-12 17:22:31,400 - MainThread ocsp_snowflake.py:588 - find_cache() - DEBUG - hit cache for subject: OrderedDict([('country_name', 'US'), ('organization_name', 'Amazon'), ('organizational_unit_name', 'Server CA 1B'), ('common_name', 'Amazon')])
2019-12-12 17:22:31,402 - MainThread ocsp_asn1crypto.py:201 - is_valid_time() - DEBUG - Verifying the attached certificate is signed by the issuer. Valid Not After: 2022-10-24 20:16:05+00:00
2019-12-12 17:22:31,403 - MainThread ocsp_snowflake.py:588 - find_cache() - DEBUG - hit cache for subject: OrderedDict([('country_name', 'US'), ('organization_name', 'Amazon'), ('common_name', 'Amazon Root CA 1')])
2019-12-12 17:22:31,406 - MainThread ocsp_asn1crypto.py:201 - is_valid_time() - DEBUG - Verifying the attached certificate is signed by the issuer. Valid Not After: 2021-12-14 07:00:00+00:00
2019-12-12 17:22:31,406 - MainThread ocsp_snowflake.py:588 - find_cache() - DEBUG - hit cache for subject: OrderedDict([('country_name', 'US'), ('state_or_province_name', 'Arizona'), ('locality_name', 'Scottsdale'), ('organization_name', 'Starfield Technologies, In
c.'), ('common_name', 'Starfield Services Root Certificate Authority - G2')])
2019-12-12 17:22:31,407 - MainThread ocsp_snowflake.py:588 - find_cache() - DEBUG - hit cache for subject: OrderedDict([('common_name', '*.snowflakecomputing.com')])
2019-12-12 17:22:31,408 - MainThread ocsp_snowflake.py:1124 - validate_by_direct_connection() - DEBUG - using OCSP response cache
2019-12-12 17:22:31,409 - MainThread ocsp_asn1crypto.py:283 - process_ocsp_response() - DEBUG - Certificate is NOT attached in Basic OCSP Response. Using issuer's certificate
2019-12-12 17:22:31,409 - MainThread ocsp_asn1crypto.py:289 - process_ocsp_response() - DEBUG - Verifying the OCSP response is signed by the issuer.
2019-12-12 17:22:31,412 - MainThread ocsp_asn1crypto.py:201 - is_valid_time() - DEBUG - Verifying the attached certificate is signed by the issuer. Valid Not After: 2022-10-24 20:13:22+00:00
2019-12-12 17:22:31,413 - MainThread ocsp_snowflake.py:588 - find_cache() - DEBUG - hit cache for subject: OrderedDict([('country_name', 'US'), ('organization_name', 'Amazon'), ('organizational_unit_name', 'Server CA 1B'), ('common_name', 'Amazon')])
2019-12-12 17:22:31,414 - MainThread ocsp_snowflake.py:1124 - validate_by_direct_connection() - DEBUG - using OCSP response cache
2019-12-12 17:22:31,415 - MainThread ocsp_asn1crypto.py:255 - process_ocsp_response() - DEBUG - Certificate is attached in Basic OCSP Response
2019-12-12 17:22:31,415 - MainThread ocsp_asn1crypto.py:257 - process_ocsp_response() - DEBUG - Verifying the attached certificate is signed by the issuer
2019-12-12 17:22:31,415 - MainThread ocsp_asn1crypto.py:261 - process_ocsp_response() - DEBUG - Valid Not After: 2022-10-24 20:13:22+00:00
2019-12-12 17:22:31,417 - MainThread ocsp_asn1crypto.py:289 - process_ocsp_response() - DEBUG - Verifying the OCSP response is signed by the issuer.
2019-12-12 17:22:31,422 - MainThread ocsp_asn1crypto.py:201 - is_valid_time() - DEBUG - Verifying the attached certificate is signed by the issuer. Valid Not After: 2022-10-24 20:16:05+00:00
2019-12-12 17:22:31,422 - MainThread ocsp_snowflake.py:588 - find_cache() - DEBUG - hit cache for subject: OrderedDict([('country_name', 'US'), ('organization_name', 'Amazon'), ('common_name', 'Amazon Root CA 1')])
2019-12-12 17:22:31,423 - MainThread ocsp_snowflake.py:1124 - validate_by_direct_connection() - DEBUG - using OCSP response cache
2019-12-12 17:22:31,424 - MainThread ocsp_asn1crypto.py:255 - process_ocsp_response() - DEBUG - Certificate is attached in Basic OCSP Response
2019-12-12 17:22:31,425 - MainThread ocsp_asn1crypto.py:257 - process_ocsp_response() - DEBUG - Verifying the attached certificate is signed by the issuer
2019-12-12 17:22:31,425 - MainThread ocsp_asn1crypto.py:261 - process_ocsp_response() - DEBUG - Valid Not After: 2022-10-24 20:16:05+00:00
2019-12-12 17:22:31,427 - MainThread ocsp_asn1crypto.py:289 - process_ocsp_response() - DEBUG - Verifying the OCSP response is signed by the issuer.
2019-12-12 17:22:31,430 - MainThread ocsp_asn1crypto.py:201 - is_valid_time() - DEBUG - Verifying the attached certificate is signed by the issuer. Valid Not After: 2021-12-14 07:00:00+00:00
2019-12-12 17:22:31,431 - MainThread ocsp_snowflake.py:588 - find_cache() - DEBUG - hit cache for subject: OrderedDict([('country_name', 'US'), ('state_or_province_name', 'Arizona'), ('locality_name', 'Scottsdale'), ('organization_name', 'Starfield Technologies, In
c.'), ('common_name', 'Starfield Services Root Certificate Authority - G2')])
2019-12-12 17:22:31,432 - MainThread ocsp_snowflake.py:1124 - validate_by_direct_connection() - DEBUG - using OCSP response cache
2019-12-12 17:22:31,433 - MainThread ocsp_asn1crypto.py:255 - process_ocsp_response() - DEBUG - Certificate is attached in Basic OCSP Response
2019-12-12 17:22:31,434 - MainThread ocsp_asn1crypto.py:257 - process_ocsp_response() - DEBUG - Verifying the attached certificate is signed by the issuer
2019-12-12 17:22:31,434 - MainThread ocsp_asn1crypto.py:261 - process_ocsp_response() - DEBUG - Valid Not After: 2021-12-14 07:00:00+00:00
2019-12-12 17:22:31,436 - MainThread ocsp_asn1crypto.py:289 - process_ocsp_response() - DEBUG - Verifying the OCSP response is signed by the issuer.
2019-12-12 17:22:31,440 - MainThread ocsp_snowflake.py:522 - write_ocsp_response_cache_file() - DEBUG - writing OCSP response cache file
2019-12-12 17:22:31,440 - MainThread ocsp_snowflake.py:1517 - encode_ocsp_response_cache() - DEBUG - encoding OCSP response cache to JSON
2019-12-12 17:22:31,450 - MainThread ocsp_snowflake.py:1015 - _validate() - DEBUG - ok
2019-12-12 17:22:31,633 - MainThread network.py:808 - _request_exec() - DEBUG - SUCCESS
2019-12-12 17:22:31,633 - MainThread network.py:953 - _use_requests_session() - DEBUG - Active requests sessions: 0, idle: 1
2019-12-12 17:22:31,633 - MainThread network.py:536 - _post_request() - DEBUG - ret[code] = None, after post request
2019-12-12 17:22:31,634 - MainThread cursor.py:543 - execute() - DEBUG - sfqid: 0190d932-02da-5955-0000-2f892c5ac20a
2019-12-12 17:22:31,634 - MainThread cursor.py:545 - execute() - INFO - query execution done
2019-12-12 17:22:31,634 - MainThread cursor.py:547 - execute() - DEBUG - SUCCESS
2019-12-12 17:22:31,635 - MainThread cursor.py:551 - execute() - DEBUG - PUT OR GET: None
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\sakkammadam\.pyenv\pyenv-win\versions\3.6.8-amd64\lib\site-packages\snowflake\connector\cursor.py", line 580, in execute
    self._init_result_and_meta(data, _use_ijson)
  File "C:\Users\sakkammadam\.pyenv\pyenv-win\versions\3.6.8-amd64\lib\site-packages\snowflake\connector\cursor.py", line 630, in _init_result_and_meta
    self._result = ArrowResult(data, self, use_dict_result=self._use_dict_result)
  File "arrow_result.pyx", line 42, in snowflake.connector.arrow_result.ArrowResult.__init__
  File "arrow_result.pyx", line 156, in snowflake.connector.arrow_result.ArrowResult._reset
NameError: name 'EmptyPyArrowIterator' is not defined

**Comment by sakkammadam** [comment]
This is also happening on Redhat 6.7

**Comment by sakkammadam** [comment]
Team - 
We noticed this on when a query containing LIMIT is supplied to certain databases.

Thanks,
Sooraj

**Comment by howryu** [comment]
Can you try the latest the 2.1.1 ? If not working, can you paste the full logs as well?

**Comment by howryu** [comment]
For 2.1.0, you can try `pip install pyarrow` and add `import pyarrow` on top of your script to see if this resolve the issue.

**Comment by RowdyHowell** [comment]
I am also experiencing the same issue running scripts that were successful earlier today.  I am simply trying to create a connection from an engine (`engine.connect()`) and get this error.

Python 3.6.7

Darwin-19.0.0-x86_64-i386-64bit

snowflake-connector-python    2.1.0
snowflake-sqlalchemy          1.1.17

**Comment by howryu** [comment]
@RowdyHowell  I am aware of an issue in 2.1.0 on macosx, can you try 2.1.1 ?

**Comment by RowdyHowell** [comment]
@howryu that worked - thank you!

**Comment by thve** [comment]
Same issue solved on Windows-10 with:

**snowflake-connector-python 2.1.1**
snowflake-sqlalchemy 1.1.17

**Comment by sakkammadam** [comment]
Team - 
Thanks for the fix - I can confirm 2.1.1 is working as intended for Windows and Redhat 6.7

If you need more info - these are the logs from a successful poll -

```
2019-12-13 08:46:59,798 - MainThread cursor.py:464 - execute() - DEBUG - executing SQL/command
DEBUG:snowflake.connector.cursor:executing SQL/command
2019-12-13 08:46:59,799 - MainThread cursor.py:487 - execute() - DEBUG - binding: [SELECT * FROM STAGE.INQUIRIES_CLN A JOIN STAGE.IAU_INQUIRIES_CLN B ON A.ID = B.I...] with input=[None], processed=[{}]
DEBUG:snowflake.connector.cursor:binding: [SELECT * FROM STAGE.INQUIRIES_CLN A JOIN STAGE.IAU_INQUIRIES_CLN B ON A.ID = B.I...] with input=[None], processed=[{}]
2019-12-13 08:46:59,799 - MainThread cursor.py:519 - execute() - INFO - query: [SELECT * FROM STAGE.INQUIRIES_CLN A JOIN STAGE.IAU_INQUIRIES_CLN B ON A.ID = B.I...]
INFO:snowflake.connector.cursor:query: [SELECT * FROM STAGE.INQUIRIES_CLN A JOIN STAGE.IAU_INQUIRIES_CLN B ON A.ID = B.I...]
2019-12-13 08:46:59,799 - MainThread connection.py:1080 - _next_sequence_counter() - DEBUG - sequence counter: 2
DEBUG:snowflake.connector.connection:sequence counter: 2
2019-12-13 08:46:59,800 - MainThread cursor.py:344 - _execute_helper() - DEBUG - running query [SELECT * FROM STAGE.INQUIRIES_CLN A JOIN STAGE.IAU_INQUIRIES_CLN B ON A.ID = B.I...]
DEBUG:snowflake.connector.cursor:running query [SELECT * FROM STAGE.INQUIRIES_CLN A JOIN STAGE.IAU_INQUIRIES_CLN B ON A.ID = B.I...]
2019-12-13 08:46:59,800 - MainThread cursor.py:353 - _execute_helper() - DEBUG - is_file_transfer: False
DEBUG:snowflake.connector.cursor:is_file_transfer: False
2019-12-13 08:46:59,800 - MainThread connection.py:836 - cmd_query() - DEBUG - _cmd_query
DEBUG:snowflake.connector.connection:_cmd_query
2019-12-13 08:46:59,800 - MainThread connection.py:858 - cmd_query() - DEBUG - sql=[SELECT * FROM STAGE.INQUIRIES_CLN A JOIN STAGE.IAU_INQUIRIES_CLN B ON A.ID = B.I...], sequence_id=[2], is_file_transfer=[None]
DEBUG:snowflake.connector.connection:sql=[SELECT * FROM STAGE.INQUIRIES_CLN A JOIN STAGE.IAU_INQUIRIES_CLN B ON A.ID = B.I...], sequence_id=[2], is_file_transfer=[None]
2019-12-13 08:46:59,800 - MainThread network.py:940 - _use_requests_session() - DEBUG - Active requests sessions: 1, idle: 0
DEBUG:snowflake.connector.network:Active requests sessions: 1, idle: 0
2019-12-13 08:46:59,800 - MainThread network.py:639 - _request_exec_wrapper() - DEBUG - remaining request timeout: None, retry cnt: 1
DEBUG:snowflake.connector.network:remaining request timeout: None, retry cnt: 1
2019-12-13 08:46:59,801 - MainThread network.py:778 - _request_exec() - DEBUG - socket timeout: 60
DEBUG:snowflake.connector.network:socket timeout: 60
2019-12-13 08:46:59,948 - MainThread network.py:808 - _request_exec() - DEBUG - SUCCESS
DEBUG:snowflake.connector.network:SUCCESS
2019-12-13 08:46:59,949 - MainThread network.py:953 - _use_requests_session() - DEBUG - Active requests sessions: 0, idle: 1
DEBUG:snowflake.connector.network:Active requests sessions: 0, idle: 1
2019-12-13 08:46:59,949 - MainThread network.py:536 - _post_request() - DEBUG - ret[code] = None, after post request
DEBUG:snowflake.connector.network:ret[code] = None, after post request
2019-12-13 08:46:59,949 - MainThread cursor.py:543 - execute() - DEBUG - sfqid: 0190dcce-0276-e0ec-0000-2f892c89c6a2
DEBUG:snowflake.connector.cursor:sfqid: 0190dcce-0276-e0ec-0000-2f892c89c6a2
2019-12-13 08:46:59,949 - MainThread cursor.py:545 - execute() - INFO - query execution done
INFO:snowflake.connector.cursor:query execution done
2019-12-13 08:46:59,949 - MainThread cursor.py:547 - execute() - DEBUG - SUCCESS
DEBUG:snowflake.connector.cursor:SUCCESS
2019-12-13 08:46:59,950 - MainThread cursor.py:551 - execute() - DEBUG - PUT OR GET: None
DEBUG:snowflake.connector.cursor:PUT OR GET: None
**2019-12-13 08:46:59,951 - MainThread json_result.py:116 - __next__() - INFO - fetching data done
INFO:snowflake.connector.json_result:fetching data done**

```

**Comment by sakkammadam** [comment]
Team - 
Thanks for the fix - I can confirm 2.1.1 is working as intended for Windows and Redhat 6.7

If you need more info - these are the logs from a successful poll -

```
2019-12-13 08:46:59,798 - MainThread cursor.py:464 - execute() - DEBUG - executing SQL/command
DEBUG:snowflake.connector.cursor:executing SQL/command
2019-12-13 08:46:59,799 - MainThread cursor.py:487 - execute() - DEBUG - binding: [SELECT * FROM STAGE.INQUIRIES_CLN A JOIN STAGE.IAU_INQUIRIES_CLN B ON A.ID = B.I...] with input=[None], processed=[{}]
DEBUG:snowflake.connector.cursor:binding: [SELECT * FROM STAGE.INQUIRIES_CLN A JOIN STAGE.IAU_INQUIRIES_CLN B ON A.ID = B.I...] with input=[None], processed=[{}]
2019-12-13 08:46:59,799 - MainThread cursor.py:519 - execute() - INFO - query: [SELECT * FROM STAGE.INQUIRIES_CLN A JOIN STAGE.IAU_INQUIRIES_CLN B ON A.ID = B.I...]
INFO:snowflake.connector.cursor:query: [SELECT * FROM STAGE.INQUIRIES_CLN A JOIN STAGE.IAU_INQUIRIES_CLN B ON A.ID = B.I...]
2019-12-13 08:46:59,799 - MainThread connection.py:1080 - _next_sequence_counter() - DEBUG - sequence counter: 2
DEBUG:snowflake.connector.connection:sequence counter: 2
2019-12-13 08:46:59,800 - MainThread cursor.py:344 - _execute_helper() - DEBUG - running query [SELECT * FROM STAGE.INQUIRIES_CLN A JOIN STAGE.IAU_INQUIRIES_CLN B ON A.ID = B.I...]
DEBUG:snowflake.connector.cursor:running query [SELECT * FROM STAGE.INQUIRIES_CLN A JOIN STAGE.IAU_INQUIRIES_CLN B ON A.ID = B.I...]
2019-12-13 08:46:59,800 - MainThread cursor.py:353 - _execute_helper() - DEBUG - is_file_transfer: False
DEBUG:snowflake.connector.cursor:is_file_transfer: False
2019-12-13 08:46:59,800 - MainThread connection.py:836 - cmd_query() - DEBUG - _cmd_query
DEBUG:snowflake.connector.connection:_cmd_query
2019-12-13 08:46:59,800 - MainThread connection.py:858 - cmd_query() - DEBUG - sql=[SELECT * FROM STAGE.INQUIRIES_CLN A JOIN STAGE.IAU_INQUIRIES_CLN B ON A.ID = B.I...], sequence_id=[2], is_file_transfer=[None]
DEBUG:snowflake.connector.connection:sql=[SELECT * FROM STAGE.INQUIRIES_CLN A JOIN STAGE.IAU_INQUIRIES_CLN B ON A.ID = B.I...], sequence_id=[2], is_file_transfer=[None]
2019-12-13 08:46:59,800 - MainThread network.py:940 - _use_requests_session() - DEBUG - Active requests sessions: 1, idle: 0
DEBUG:snowflake.connector.network:Active requests sessions: 1, idle: 0
2019-12-13 08:46:59,800 - MainThread network.py:639 - _request_exec_wrapper() - DEBUG - remaining request timeout: None, retry cnt: 1
DEBUG:snowflake.connector.network:remaining request timeout: None, retry cnt: 1
2019-12-13 08:46:59,801 - MainThread network.py:778 - _request_exec() - DEBUG - socket timeout: 60
DEBUG:snowflake.connector.network:socket timeout: 60
2019-12-13 08:46:59,948 - MainThread network.py:808 - _request_exec() - DEBUG - SUCCESS
DEBUG:snowflake.connector.network:SUCCESS
2019-12-13 08:46:59,949 - MainThread network.py:953 - _use_requests_session() - DEBUG - Active requests sessions: 0, idle: 1
DEBUG:snowflake.connector.network:Active requests sessions: 0, idle: 1
2019-12-13 08:46:59,949 - MainThread network.py:536 - _post_request() - DEBUG - ret[code] = None, after post request
DEBUG:snowflake.connector.network:ret[code] = None, after post request
2019-12-13 08:46:59,949 - MainThread cursor.py:543 - execute() - DEBUG - sfqid: 0190dcce-0276-e0ec-0000-2f892c89c6a2
DEBUG:snowflake.connector.cursor:sfqid: 0190dcce-0276-e0ec-0000-2f892c89c6a2
2019-12-13 08:46:59,949 - MainThread cursor.py:545 - execute() - INFO - query execution done
INFO:snowflake.connector.cursor:query execution done
2019-12-13 08:46:59,949 - MainThread cursor.py:547 - execute() - DEBUG - SUCCESS
DEBUG:snowflake.connector.cursor:SUCCESS
2019-12-13 08:46:59,950 - MainThread cursor.py:551 - execute() - DEBUG - PUT OR GET: None
DEBUG:snowflake.connector.cursor:PUT OR GET: None
**2019-12-13 08:46:59,951 - MainThread json_result.py:116 - __next__() - INFO - fetching data done
INFO:snowflake.connector.json_result:fetching data done**

```

**Comment by sakkammadam** [comment]
Team - 
Thanks for the fix - I can confirm 2.1.1 is working as intended for Windows and Redhat 6.7

If you need more info - these are the logs from a successful poll -

```
2019-12-13 08:46:59,798 - MainThread cursor.py:464 - execute() - DEBUG - executing SQL/command
DEBUG:snowflake.connector.cursor:executing SQL/command
2019-12-13 08:46:59,799 - MainThread cursor.py:487 - execute() - DEBUG - binding: [SELECT * FROM STAGE.INQUIRIES_CLN A JOIN STAGE.IAU_INQUIRIES_CLN B ON A.ID = B.I...] with input=[None], processed=[{}]
DEBUG:snowflake.connector.cursor:binding: [SELECT * FROM STAGE.INQUIRIES_CLN A JOIN STAGE.IAU_INQUIRIES_CLN B ON A.ID = B.I...] with input=[None], processed=[{}]
2019-12-13 08:46:59,799 - MainThread cursor.py:519 - execute() - INFO - query: [SELECT * FROM STAGE.INQUIRIES_CLN A JOIN STAGE.IAU_INQUIRIES_CLN B ON A.ID = B.I...]
INFO:snowflake.connector.cursor:query: [SELECT * FROM STAGE.INQUIRIES_CLN A JOIN STAGE.IAU_INQUIRIES_CLN B ON A.ID = B.I...]
2019-12-13 08:46:59,799 - MainThread connection.py:1080 - _next_sequence_counter() - DEBUG - sequence counter: 2
DEBUG:snowflake.connector.connection:sequence counter: 2
2019-12-13 08:46:59,800 - MainThread cursor.py:344 - _execute_helper() - DEBUG - running query [SELECT * FROM STAGE.INQUIRIES_CLN A JOIN STAGE.IAU_INQUIRIES_CLN B ON A.ID = B.I...]
DEBUG:snowflake.connector.cursor:running query [SELECT * FROM STAGE.INQUIRIES_CLN A JOIN STAGE.IAU_INQUIRIES_CLN B ON A.ID = B.I...]
2019-12-13 08:46:59,800 - MainThread cursor.py:353 - _execute_helper() - DEBUG - is_file_transfer: False
DEBUG:snowflake.connector.cursor:is_file_transfer: False
2019-12-13 08:46:59,800 - MainThread connection.py:836 - cmd_query() - DEBUG - _cmd_query
DEBUG:snowflake.connector.connection:_cmd_query
2019-12-13 08:46:59,800 - MainThread connection.py:858 - cmd_query() - DEBUG - sql=[SELECT * FROM STAGE.INQUIRIES_CLN A JOIN STAGE.IAU_INQUIRIES_CLN B ON A.ID = B.I...], sequence_id=[2], is_file_transfer=[None]
DEBUG:snowflake.connector.connection:sql=[SELECT * FROM STAGE.INQUIRIES_CLN A JOIN STAGE.IAU_INQUIRIES_CLN B ON A.ID = B.I...], sequence_id=[2], is_file_transfer=[None]
2019-12-13 08:46:59,800 - MainThread network.py:940 - _use_requests_session() - DEBUG - Active requests sessions: 1, idle: 0
DEBUG:snowflake.connector.network:Active requests sessions: 1, idle: 0
2019-12-13 08:46:59,800 - MainThread network.py:639 - _request_exec_wrapper() - DEBUG - remaining request timeout: None, retry cnt: 1
DEBUG:snowflake.connector.network:remaining request timeout: None, retry cnt: 1
2019-12-13 08:46:59,801 - MainThread network.py:778 - _request_exec() - DEBUG - socket timeout: 60
DEBUG:snowflake.connector.network:socket timeout: 60
2019-12-13 08:46:59,948 - MainThread network.py:808 - _request_exec() - DEBUG - SUCCESS
DEBUG:snowflake.connector.network:SUCCESS
2019-12-13 08:46:59,949 - MainThread network.py:953 - _use_requests_session() - DEBUG - Active requests sessions: 0, idle: 1
DEBUG:snowflake.connector.network:Active requests sessions: 0, idle: 1
2019-12-13 08:46:59,949 - MainThread network.py:536 - _post_request() - DEBUG - ret[code] = None, after post request
DEBUG:snowflake.connector.network:ret[code] = None, after post request
2019-12-13 08:46:59,949 - MainThread cursor.py:543 - execute() - DEBUG - sfqid: 0190dcce-0276-e0ec-0000-2f892c89c6a2
DEBUG:snowflake.connector.cursor:sfqid: 0190dcce-0276-e0ec-0000-2f892c89c6a2
2019-12-13 08:46:59,949 - MainThread cursor.py:545 - execute() - INFO - query execution done
INFO:snowflake.connector.cursor:query execution done
2019-12-13 08:46:59,949 - MainThread cursor.py:547 - execute() - DEBUG - SUCCESS
DEBUG:snowflake.connector.cursor:SUCCESS
2019-12-13 08:46:59,950 - MainThread cursor.py:551 - execute() - DEBUG - PUT OR GET: None
DEBUG:snowflake.connector.cursor:PUT OR GET: None
**2019-12-13 08:46:59,951 - MainThread json_result.py:116 - __next__() - INFO - fetching data done
INFO:snowflake.connector.json_result:fetching data done**

```
